### DISCUSSION TITLE
弹幕搜索无结果是什么原因

### DISCUSSION THREAD

**Discussion by Minsia** [Tony15246/uosc_danmaku#82] [Q&A] [answered]
检查了杀软没有查杀DanmakuFactory.exe
![9fc5c4e2-2a12-44a2-a3ca-db8df274c55c](https://github.com/user-attachments/assets/bcec477f-313a-4eeb-a4bc-e87c53442e4f)

**Comment by Tony15246** [comment]
控制台有报错吗？

**Reply by Tony15246** [reply]
使用的是release v1.2.0版或者是最新版源代码吗？如果不是的话，可以先更新到最新版试试看。

如果是release v1.2.0版或者是最新版源代码，并且控制台也没有明显报错的话，把mpv的运行日志发给我一下，我帮你看看是什么问题。

**Reply by Minsia** [reply]
1.2版本的控制台
![image](https://github.com/user-attachments/assets/6921c823-8e98-4355-88c5-e23409b51174)

**Reply by Tony15246** [reply]
向弹弹play发送请求的curl命令失败了，网络有问题。你配置过`proxy`代理选项吗？配置过的话把`proxy`代理的配置先删掉再试一下。

你可以在系统命令行运行一下下面的命令验证一下你的网络环境下能否正常请求弹弹play的api

``` shell
curl -X GET --header 'Accept: application/json' 'https://api.dandanplay.net/api/v2/search/episodes?anime=%E9%85%B7%E7%88%B1%E7%94%B5%E5%BD%B1%E7%9A%84'
```

**Reply by Tony15246** [reply]
另外，你在系统的命令行运行一下下面这个命令看看，我确认一下是不是你本地的curl工具版本太老旧导致的问题

```
curl --version
```

**Reply by Minsia** [reply]
![f881c6402b1e83e728a95a65aba042d3](https://github.com/user-attachments/assets/9608c42a-7cca-45c9-af15-6991870ad090)
![da2a6d8c7ab86bd8b27ca6a7a060b32e](https://github.com/user-attachments/assets/8cff6016-1304-4e64-8dee-944ed9f75a97)
提示我没装，这个系统不是自带的吗

**Reply by Tony15246** [reply]
> ![da2a6d8c7ab86bd8b27ca6a7a060b32e](https://github.com/user-attachments/assets/8cff6016-1304-4e64-8dee-944ed9f75a97)
> 提示我没装，这个系统不是自带的吗

这个正常,mpv的控制台肯定是不能执行系统命令的，只能执行mpv支持的命令。

> ![f881c6402b1e83e728a95a65aba042d3](https://github.com/user-attachments/assets/9608c42a-7cca-45c9-af15-6991870ad090)

你把这个命令中的单引号替换成双引号重新执行一下试试，运行结果截图给我看一下

``` shell
curl -X GET --header "Accept: application/json" "https://api.dandanplay.net/api/v2/search/episodes?anime=%E9%85%B7%E7%88%B1%E7%94%B5%E5%BD%B1%E7%9A%84"
```

**Reply by Minsia** [reply]
![image](https://github.com/user-attachments/assets/03c1be7e-17c0-4bf3-84fb-4775afea3311)

**Reply by Tony15246** [reply]
你的系统不知道为什么无法访问证书吊销检查服务器，导致系统在验证 HTTPS 连接的 SSL/TLS 证书时，无法检查证书吊销状态。

你尝试一下添加`--ssl-no-revoke`参数强行禁用证书吊销检查，看看能否正常请求，运行结果截图给我看一下还有没有报错

``` shell
curl -X GET --ssl-no-revoke --header "Accept: application/json" "https://api.dandanplay.net/api/v2/search/episodes?anime=%E9%85%B7%E7%88%B1%E7%94%B5%E5%BD%B1%E7%9A%84"
```

**Reply by Minsia** [reply]
![image](https://github.com/user-attachments/assets/fd03fc67-537a-48a1-83ae-a393f8bff03e)

**Reply by Tony15246** [reply]
这么看来跳过证书吊销检查就能正常请求了

**Reply by Tony15246** [reply]
你到系统的`C:\Users\<用户名>\`目录下，创建一个文件，命名为`.curlrc`，文件内容就写`--ssl-no-revoke`，写完保存退出就行

然后在命令行里重新试一下，看看现在能不能正常执行下面这个命令了

``` shell
curl -X GET --header "Accept: application/json" "https://api.dandanplay.net/api/v2/search/episodes?anime=%E9%85%B7%E7%88%B1%E7%94%B5%E5%BD%B1%E7%9A%84"
```

如果执行完的结果和下面这张图里一样，那就可以去试试看插件现在能不能正常搜索弹幕了

![image](https://github.com/user-attachments/assets/fd03fc67-537a-48a1-83ae-a393f8bff03e)

**Reply by Minsia** [reply]
创建完成后执行还是无法检查

**Reply by Tony15246** [reply]
竟然不能生效😂可恶的windows，配置文件管理机制乱七八糟，十有八九又是被哪里的配置覆盖掉了

**Reply by Tony15246** [reply]
你本地环境配不好的话那只能修改源代码，强行禁用证书吊销检查了

**Reply by Tony15246** [reply]
找到插件所在的文件夹，修改`api.lua`，找到这一行

https://github.com/Tony15246/uosc_danmaku/blob/04f4b5c206f069260ae92f65b3f029fefea9ca5d/api.lua#L434

改成下面这样

``` lua
"curl", "--ssl-no-revoke", "-s", "-X", "POST", url, 
```

**Reply by Tony15246** [reply]
找到这一段

https://github.com/Tony15246/uosc_danmaku/blob/04f4b5c206f069260ae92f65b3f029fefea9ca5d/api.lua#L463-L474

把对应部分改成下面这样

``` lua
 "curl", 
 "--ssl-no-revoke",
 "--range", 
```

**Reply by Tony15246** [reply]
找到这一段

https://github.com/Tony15246/uosc_danmaku/blob/04f4b5c206f069260ae92f65b3f029fefea9ca5d/api.lua#L493-L505

把对应部分改成下面这样

```lua
 "curl", 
 "--ssl-no-revoke",
 "-L", 
```

**Reply by Tony15246** [reply]
找到这一段

https://github.com/Tony15246/uosc_danmaku/blob/04f4b5c206f069260ae92f65b3f029fefea9ca5d/api.lua#L976-L986

把对应部分改成下面这样

``` lua
 "curl", 
 "--ssl-no-revoke",
 "-L", 
```

**Reply by Tony15246** [reply]
找到这一段

https://github.com/Tony15246/uosc_danmaku/blob/04f4b5c206f069260ae92f65b3f029fefea9ca5d/api.lua#L1012-L1033

把对应部分改成下面这样

``` lua
     "curl", 
     "--ssl-no-revoke",
     "-X", 
     "POST", 
```

**Reply by Tony15246** [reply, marked as answer]
一共这五处地方，总之就是找到使用curl的地方，全都添上`--ssl-no-revoke`参数

然后你再试试用一下插件，应该就能正常搜索弹幕了

**Reply by Tony15246** [reply]
我会考虑在后续的更新里加入对curl设置ssl-no-revoke参数的支持或自定义参数的支持

**Reply by Minsia** [reply]
![image](https://github.com/user-attachments/assets/90f0a5f9-d3c3-4479-b8bf-6d50ff5a932c)
改完了弹幕控件会变成这样

**Reply by Tony15246** [reply]
控制台有报错吗，有报错的话截图给我看一下。可能是改的过程中出了某些问题，导致插件在初始化控件之前就出错了从而没有正确初始化弹幕控件

**Reply by Tony15246** [reply]
找不到问题的话可以把你改过的`api.lua`和mpv的运行日志发给我，我帮你排查一下

**Reply by Minsia** [reply]
[api、log.zip](https://github.com/user-attachments/files/17935055/api.log.zip)
耽搁了点时间不好意思，没找出问题在哪

**Reply by Tony15246** [reply]
没事，我看一下

**Reply by Tony15246** [reply]
是我没表述清楚，我的问题。我说的`对应部分改成下面这样`意思不是直接替换，而是要把`--ssl-no-revoke`插入进去的意思，插入完以后会变成下面那样😂

没事儿，你直接用我改好的`api.lua`覆盖替换一下吧

[api.zip](https://github.com/user-attachments/files/17935422/api.zip)

**Reply by Minsia** [reply]
我的问题，没注意看后面是找对应的，换完现在可以了，感谢😂

**Reply by Tony15246** [reply]
没事，不用谢(ゝ∀･)

**Comment by limqiaolezi** [comment]
<img width="2560" height="1440" alt="image" src="https://github.com/user-attachments/assets/6d691a7c-7b24-4a4f-865e-5396a11f5201" />
大佬我也想问下遇到HTTP请求失败的问题，之前是可以搜索到的今天突然不行了，用的1.3.1版本+mpv_lazy

**Reply by Minsia** [reply]
你好像只回复给我了，我之前是按照上面把api里的所有"curl", 
后面加上 "--ssl-no-revoke"就可以了， 刚刚想看看报错突然发现不用改也行了，帮你@一下吧

**Reply by Minsia** [reply]
@Tony15246 这里有弹幕无法加载的问题

**Reply by limqiaolezi** [reply]
感谢！

**Reply by Tony15246** [reply]
抱歉之前没看到这个comment，这两天沉迷玩《死亡搁浅》赛博送快递，没查看邮箱

这个图片里的信息不太够，可以提供一下出现问题时的mpv的运行日志吗？另外这个HTTP请求失败的问题是看每个视频都会出现，还是看个别动漫或者视频的时候会出现？

**Reply by limqiaolezi** [reply]
[mpv-lazy.log](https://github.com/user-attachments/files/21232747/mpv-lazy.log)
大佬我刚回到家，今天试了一下，还是都没法获取弹幕，还希望能麻烦您看看！

**Reply by Tony15246** [reply]
> [mpv-lazy.log](https://github.com/user-attachments/files/21232747/mpv-lazy.log) 大佬我刚回到家，今天试了一下，还是都没法获取弹幕，还希望能麻烦您看看！

奇怪，日志里也没有明显的报错，还是只有一个子进程执行失败的消息。线索有点少，我需要缩小一下问题范围。

我看日志里是在反复执行开关弹幕操作，相当于重复执行自动弹幕加载。如果改用手动搜索的话能够成功加载弹幕吗？

**Reply by limqiaolezi** [reply]
手动也无法搜索到任何番剧
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/5b963e1a-7ffa-429e-81d5-f721af13906f" />

**Reply by MiKoto-Railgun** [reply]
报错看着怎么像路径问题，要不你管理员模式启动mpv试试

**Reply by limqiaolezi** [reply]
[mpv-lazy.log](https://github.com/user-attachments/files/21236620/mpv-lazy.log)
以管理员模式启动也没法加载弹幕

**Reply by limqiaolezi** [reply]
大佬，我好像发现了原因
确实是路径问题，”~uosc_danmaku\bin\dandanplay“少了dandanplay.exe，而且运行后会自动删除，检查了一下应该是window安全中心自动扫描删除的

**Reply by Tony15246** [reply]
> 大佬，我好像发现了原因 确实是路径问题，”~uosc_danmaku\bin\dandanplay“少了dandanplay.exe，而且运行后会自动删除，检查了一下应该是window安全中心自动扫描删除的

我有想过是不是dandanplay.exe被windows查杀了，但是从日志里又看不出来端倪，没想到还真是。

又又又是windows安全中心搞的鬼。我真是服了windows了，一个操作系统天天替用户做决定，管的也太宽了。一起加入linux大家庭吧😆

**Reply by hooke007** [reply]
加段必要的代码检查路径文件是否存在

**Reply by Tony15246** [reply]
> 加段必要的代码检查路径文件是否存在

有道理，我加一下

**Comment by platinum2333** [comment]
name: 弹幕搜索无结果

**运行环境**
 - 操作系统: window10
 - mpv版本 [e.g. 0.40.0]
 - 是否安装uosc  是，我使用的是uosc菜单
 - 本插件版本 [e.g. v1.3.2/Lastest Release/main分支最新源代码版本]
<img width="582" height="258" alt="image" src="https://github.com/user-attachments/assets/d6fb9126-4448-4ae8-939e-7ad7e3edca5a" />
<img width="1924" height="1140" alt="image" src="https://github.com/user-attachments/assets/aedf6e5a-2be8-42fa-a98f-af739c722800" />
[log.txt](https://github.com/user-attachments/files/30310070/log.txt)

**Reply by platinum2333** [reply]
我使用的是mpv_PlayKit最新版，从uosc用的是最新版，界面改键没问题但是搜索不到弹幕
